dexter/src/main/java/com/karumi/dexter/listener/threaddecorator/MultiplePermissionListenerThreadDecorator.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright (C) 2015 Karumi.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.karumi.dexter.listener.threaddecorator;
+
+import com.karumi.dexter.MultiplePermissionsReport;
+import com.karumi.dexter.PermissionToken;
+import com.karumi.dexter.listener.PermissionRequest;
+import com.karumi.dexter.listener.multi.MultiplePermissionsListener;
+import java.util.List;
+
+/**
+ * Decorator to execute the permission updates on a given thread specification
+ */
+public class MultiplePermissionListenerThreadDecorator implements MultiplePermissionsListener {
```

Comment:
@pedrovgs what do you suggest then? The only reason it's public is because its package is not the same as the `DexterInstance` once, where the decorator is used. I could just move the decorator to the `DexterInstance` package.

The same issue happens with other classes situated under the `threaddecorator` package, such as: `ThreadSpecFactory` and `ThreadSpec`.

Let me know how you'd like to approach it.